### ISSUE TITLE
Add a splash screen to a .NET MAUI app project on per-platform approaches

### ISSUE THREAD

**Issue by JoacimWall** [dotnet/maui#12917]
### Description

In the documentation it says

For more advanced splash screen scenarios, per-platform approaches apply.

When I try to add per platform like in Xamarin the android approach works but not the IOS

get this error.

/Users/joacimwall/Downloads/emptyproj/emptyproj/Platforms/iOS/MauiSplashCustom.storyboard: Error: The file 'Platforms/iOS/MauiSplashCustom.storyboard' conflicts with 'Platforms/iOS/MauiSplashCustom.storyboard'.
         (emptyproj)

### Steps to Reproduce

create a new maui app
remove the splash folder
add a storybord to the platforms/iOS folder 
add the key 
 <key>UILaunchStoryboardName</key>
  <string>MauiSplash</string>
to plist file.

result in build error.

/Users/joacimwall/Downloads/emptyproj/emptyproj/Platforms/iOS/MauiSplashCustom.storyboard: Error: The file 'Platforms/iOS/MauiSplashCustom.storyboard' conflicts with 'Platforms/iOS/MauiSplashCustom.storyboard'.
         (emptyproj)   

### Link to public reproduction project repository

not needed

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

all

### Did you find any workaround?

No

> Hi @JoacimWall. We have added the "s/needs-repro" label to this issue, which indicates that we require steps and sample code to reproduce the issue before we can take further action. Please try to create a minimal sample project/solution or code samples which reproduce the issue, ideally as a GitHub repo that we can clone. See more details about creating repros here: https://github.com/dotnet/maui/blob/main/.github/repro.md
> 
> This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

Repo
https://github.com/JoacimWall/Joacim_Bug_Report/tree/main/LaunchScreenIos

**Comment by mapsouza** [comment]
I was able to create the splashscreen for iOS with the following steps:

1) I placed the storyboard in the root folder of the project.
2) Set the storyboard's BuildAction property to InterfaceDefinition.
2) I put the images inside the Resources\Images folder
3) In csproj, I include these images in a conditional ItemGroup for 'net7.0-ios' only.
4) Edit the info.plist and include the statement 
<key>UILaunchStoryboardName</key>
<string>customlaunchscreen.storyboard</string>

This worked for me.

**Comment by JoacimWall** [comment]
> Hi @JoacimWall. We have added the "s/needs-repro" label to this issue, which indicates that we require steps and sample code to reproduce the issue before we can take further action. Please try to create a minimal sample project/solution or code samples which reproduce the issue, ideally as a GitHub repo that we can clone. See more details about creating repros here: https://github.com/dotnet/maui/blob/main/.github/repro.md
> 
> This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

Repo
https://github.com/JoacimWall/Joacim_Bug_Report/tree/main/LaunchScreenIos

**Comment by JoacimWall** [comment]
> 

Hi and thanks 
I try to follow you steps but I can't get the images to show. I see that the storybord is used when change background color on it. 

Is ut possible for you to share you storybord so I can se how you map you images.

Did a test project where you can see how I implemented it. 
https://github.com/JoacimWall/Joacim_Bug_Report/tree/main/LaunchScreenMaui

**Comment by mapsouza** [comment]
Hi...

This is my storyboard:

```xaml
<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="21225" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" launchScreen="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="01J-lp-oVM">
    <device id="retina6_7" orientation="portrait" appearance="light"/>
    <dependencies>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="21207"/>
        <capability name="Safe area layout guides" minToolsVersion="9.0"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <scenes>
        <!--View Controller-->
        <scene sceneID="EHf-IW-A2E">
            <objects>
                <viewController id="01J-lp-oVM" sceneMemberID="viewController">
                    <view key="view" contentMode="scaleToFill" id="Ze5-6b-2t3">
                        <rect key="frame" x="0.0" y="0.0" width="428" height="926"/>
                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                        <subviews>
                            <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleToFill" horizontalHuggingPriority="251" verticalHuggingPriority="251" fixedFrame="YES" insetsLayoutMarginsFromSafeArea="NO" image="splashscreen" translatesAutoresizingMaskIntoConstraints="NO" id="6St-vW-eOf">
                                <rect key="frame" x="0.0" y="0.0" width="428" height="926"/>
                                <autoresizingMask key="autoresizingMask" flexibleMinX="YES" widthSizable="YES" flexibleMaxX="YES" flexibleMinY="YES" heightSizable="YES" flexibleMaxY="YES"/>
                                <accessibility key="accessibilityConfiguration">
                                    <accessibilityTraits key="traits" image="YES" notEnabled="YES"/>
                                </accessibility>
                            </imageView>
                            <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFit" image="logosplash" id="WTi-Z3-dfl">
                                <rect key="frame" x="7" y="1" width="411" height="926"/>
                                <autoresizingMask key="autoresizingMask" flexibleMinX="YES" widthSizable="YES" flexibleMaxX="YES" flexibleMinY="YES" heightSizable="YES" flexibleMaxY="YES"/>
                            </imageView>
                        </subviews>
                        <viewLayoutGuide key="safeArea" id="Bcu-3y-fUS"/>
                        <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
                    </view>
                </viewController>
                <placeholder placeholderIdentifier="IBFirstResponder" id="iYj-Kq-Ea1" userLabel="First Responder" sceneMemberID="firstResponder"/>
            </objects>
            <point key="canvasLocation" x="52.307692307692307" y="374.64454976303318"/>
        </scene>
    </scenes>
    <resources>
        <image name="logosplashscreen" width="413.33334350585938" height="160"/>
        <image name="splashscreen" width="413.33334350585938" height="736"/>
    </resources>
</document>
```
and my csproj...

````xml
	<ItemGroup Condition="'$(TargetFramework)'=='net7.0-ios'">
		<EmbeddedResource Include="Resources\Images\logosplash.png" />
		<EmbeddedResource Include="Resources\Images\splashscreen.png" />
	</ItemGroup>
````

I am confused now.

- In my project tree in VS 2022, the storyboard is at the root of the project.

<img width="244" alt="image" src="https://user-images.githubusercontent.com/31948355/215093161-2a50bbd6-79d4-4d7c-b961-f430c9a475ca.png">

- When I search the file on Github, the file was inside Platforms/iOS

I have no storyboard references in my csproj. Really I don't understand how it's located this way.

By the way, when I did it, I was using my iPhone to test. In each change, you need to delete the application, clean the project and restart the device to recognize the changes. It seems that this is already an old bug of Apple devices.

**Comment by jfversluis** [comment]
I'm not really clear on the status of this issue. Either way, this seems like more of a "how to" discussion than an actual issue in .NET MAUI. You definitely should be able to still use a custom splash screen.

Might be a nice candidate for an addition to the docs @davidbritch.

At this point I'm inclined to think that the functionality should work, but it's unclear how to do it. I think it might be better to open a discussion and/or maybe ask on a medium like Stack Overflow. 

If you do think something is actually not working and thus a bug with .NET MAUI, feel free to open a new issue.

Thanks!

**Comment by JoacimWall** [comment]
So these are my experiences after first doing this on a Mac with VS preview.

1. don't put storyboard in the logic place Platform/Ios you will get this error.
  The file 'Platforms/iOS/LaunchScreen.storyboard' conflicts with 'Platforms/iOS/LaunchScreen.storyboard'.
          (LaunchScreenMAUI)
Use the root of the project.

2. in your info.plist do not do as in Xamarin and enter <key>UILaunchStoryboardName</key>
     <string>LaunchScreen</string>
you need to add the full name
    <string>LaunchScreen.storyboard</string>

3. Delete the Splash folder under resources otherwise it will try to add Mauisplashscreen

4. add your images as png not svg and they should be as below in your project file.
<ItemGroup Condition="'$(TargetFramework)'=='net7.0-ios'">
<BundleResource Include="Resources\Images\splashbkg.png" />
<BundleResource Include="Resources\Images\splashtext.png" />
</ItemGroup>

here is a repo if you want to test/copy the logic
https://github.com/JoacimWall/Joacim_Bug_Report/tree/main/LaunchScreenMaui

This one liner in the docs for MAUI
"For more advanced splash screen scenarios, per-platform approaches apply." Need some updates.

finally thanks mapsouza for all the help.

**Comment by mapsouza** [comment]
I think on it like a bug. In my opinion, storyboard file needs to stay in platforms/iOS folder. But when we try this, the error occurs. Sorry because my workaround generate a discussion! 😉

**Comment by JoacimWall** [comment]
Hi 
if you try to build the same project that now works on Mac on a Windows machine in latest Visual studio you get this error. 
Repo
https://github.com/JoacimWall/Joacim_Bug_Report/tree/main/LaunchScreenMaui


1>Resources\Images\splashbkg.png : error : The name 'Resources' is reserved and cannot be used.
1>Resources\Images\splashbkg.png : error :
1>Resources\Images\splashtext.png : error : The name 'Resources' is reserved and cannot be used.
1>Resources\Images\splashtext.png : error :
1>Done building project "LaunchScreenMAUI.csproj" -- FAILED.

So I say this looks like bug to me.

//Joacim

**Comment by breenbob** [comment]
Thanks @JoacimWall and @mapsouza - your guidance helped me with same issue. I also put the storyboard file in /Platforms/iOS folder thinking it was the logical place to put it. So definitely agree it would help if docs were updated.

**Comment by duongnb-runsystem** [comment]
@JoacimWall  tks you, above is for iOS, don't know if you have tried on android yet?
